### PR TITLE
Issue 80: Adding Helm Charts for Zookeper deployment and testing

### LINKED ISSUE #80: Zookeeper health check
This is to add Helm Charts to Zookeeper so as to implement `helm test` on it so that post an upgrade we can determine whether the upgrade was successful or not by simply running a `helm test <release-name>`

### PR DESCRIPTION
**Pull request by SrishT**
Signed-off-by: SrishT <Srishti.Thakkar@dell.com>

### Change log description
- Adding helm chart to deploy zookeeper-operator and zookeeper into Kubernetes cluster
- Implementing helm test to test successful deployment/upgrade of zookeeper cluster 

### Purpose of the change
Fixes #80 

### How to verify it
To deploy zookeper operator
`helm install charts/zookeeper-operator --name <zk-operator-release-name>`

To deploy zookeper cluster
`helm install charts/zookeeper --name <zk-release-name>`

To test the zookeeper cluster deployment/upgrade
`helm test <zk-release-name>`

If the zookeeper deployment/upgrade was successful and all pods are in Ready state then the test should pass, else it should fail.

### CODE REVIEW

**Comment by jdmaguire** [comment]
A bit ignorant here. What is the difference between this and say https://github.com/pravega/charts (note, private Github repo)?

**Comment by jdmaguire** [comment]
I am a bit ignorant here. What is the difference between this and say https://github.com/pravega/charts (note, private Github repo)?

**Comment by tewit** [comment]
LGTM

File: charts/zookeeper-operator/templates/clusterrole.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: {{ template "zookeeper-operator.fullname" . }}
+rules:
+- apiGroups:
+  - zookeeper.pravega.io
+  resources:
+  - "*"
+  verbs:
+  - "*"
```
Comment (pbelgundi) [inline charts/zookeeper-operator/templates/clusterrole.yaml]: Same as earlier helm test. Can we not do with only `get` under 'verbs` here?

Comment (SrishT) [reply charts/zookeeper-operator/templates/clusterrole.yaml]: This piece of code is for granting permissions for zookeeper operator deployment and not for running the test.

File: charts/zookeeper-operator/Chart.yaml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+name: zookeeper-operator
+version: 0.2.2
+appVersion: 0.2.2
+description: zookeeper operator Helm chart for Kubernetes
+keywords:
+- zookeeper
+- storage
+home: https://github.com/pravega/charts/blob/master/zookeeper-operator
```
Comment (pbelgundi) [inline charts/zookeeper-operator/Chart.yaml]: This should point to charts under zookeeper-operator

File: charts/zookeeper-operator/Chart.yaml [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+icon: https://zookeeper.apache.org/images/zookeeper_small.gif
+sources:
```
Comment (pbelgundi) [inline charts/zookeeper-operator/Chart.yaml]: Same as `home`

File: charts/zookeeper-operator/Chart.yaml (hunk 2 above)
Comment (SrishT) [reply charts/zookeeper-operator/Chart.yaml]: done

File: charts/zookeeper-operator/Chart.yaml (hunk 3 above)
Comment (SrishT) [reply charts/zookeeper-operator/Chart.yaml]: done

**Pull request merged**